packages/terra-divider/src/Divider.jsx

```diff
@@ -1,16 +1,37 @@
 import React from 'react';
 import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
 import styles from './Divider.module.scss';
 
+const propTypes = {
+  /**
+  * String to be displayed in-line with the divider.
+  */
+  text: PropTypes.string,
+};
+
 const cx = classNames.bind(styles);
 
-const Divider = ({ ...customProps }) => {
+const Divider = (props) => {
+  const { text, ...customProps } = props;
+
   const dividerClassNames = cx([
     'divider',
     customProps.className,
   ]);
 
-  return <hr {...customProps} className={dividerClassNames} aria-hidden="true" />;
+  const dividerType = !text
+    ? <hr {...customProps} className={dividerClassNames} aria-hidden="true" />
+    : (
+      <div className={cx(['divider-container'])}>
+        <span className={cx(['divider-text'])}>{text}</span>
+      </div>
+    );
+
+  return (
+    dividerType
+  );
```

Comment:
This is up to your preference, leave it as is if you prefer it this way.  

Personally I think it's easier to read components without ternaries within the render logic.


```jsx
  if (!text) {
    return <hr {...customProps} className={dividerClassNames} aria-hidden="true" />;
  }

  return (
    <div className={cx(['divider-container'])}>
      <span className={cx(['divider-text'])}>{text}</span>
    </div>
  );
```

Or

```jsx
  if (text) {
    return (
      <div className={cx(['divider-container'])}>
        <span className={cx(['divider-text'])}>{text}</span>
      </div>
    );
  }

  return <hr {...customProps} className={dividerClassNames} aria-hidden="true" />;
```

Reply:
Honestly I don’t really have a preference. Either the ternary or the suggestions above are good for me, so if it makes it easier to parse for others, I’m good with changing it.

Reply:
Changes added! [7ad1cc8](https://github.com/cerner/terra-core/pull/2296/commits/7ad1cc8ca8af947a4b40c5cbf02c9a0bf49c534f)